### PR TITLE
Use table to render data

### LINKED ISSUE #18: Use text-table in verbose mode
What about using https://www.npmjs.com/package/text-table in -v?

### PR DESCRIPTION
**Pull request by DylanPiercey**
This should resolve https://github.com/antonmedv/jsize/issues/18 (checkout both normal and verbose mode).

I messed with it for quite a while to get it to look decent, let me know what you think or if you have any ideas.

### CODE REVIEW

File: bin/jsize [hunk 1]
```diff
@@ -7,61 +7,119 @@ const pkg = require('../package.json')
...
 program
   .usage('[options] <name ...>')
   .option('-v, --verbose', 'Display initial, minified and gzipped size.')
   .parse(process.argv)
 
-// Calucate the padding based on the longest package name.
-const padding = 4 + program.args.reduce(maxLength, 0)
+// Store results for rendering later.
+const bundles = []
+const headings = ['.', '.']
```
Comment (antonmedv) [inline bin/jsize]: What about `['package', '']`?

File: bin/jsize [hunk 2]
```diff
@@ -7,61 +7,119 @@ const pkg = require('../package.json')
...
+  .map(it => chalk.hidden(it))
+  .concat(['initial', 'minified', 'gzipped'].map(
+    heading => chalk.bold.dim(heading)
```
Comment (antonmedv) [inline bin/jsize]: I think bold dim is king of strange, maybe leave only dim?

File: bin/jsize [hunk 3]
```diff
@@ -7,61 +7,119 @@ const pkg = require('../package.json')
...
-  let info = ''
   if (program.verbose) {
-    info += padEnd(prettyBytes(result.initial) + ' (initial)', 20, ' ')
-    info += padEnd(prettyBytes(result.minified) + ' (minified)', 20, ' ')
+    table.unshift(headings, padding)
   }
-  info += colorBytes(result.gzipped) + ' (gzipped)'
-  spinner.succeed(padEnd(spinner.text, padding, ' ') + info)
+
+  // Add padding to table.
```
Comment (antonmedv) [inline bin/jsize]: What about adding padding from left? Two spaces for each row – will be looking better.

File: bin/jsize (hunk 1 above)
Comment (DylanPiercey) [reply bin/jsize]: I think it looks strange with package personally.

File: bin/jsize (hunk 2 above)
Comment (DylanPiercey) [reply bin/jsize]: Can do.

File: bin/jsize (hunk 3 above)
Comment (DylanPiercey) [reply bin/jsize]: I'll try it out.

**Comment by DylanPiercey** [comment]
@antonmedv I ended up adding the 'package' header, uppercased the headers and removed the bold.

I tried adding the padding to the left however it strangely aligned the whole table :(.

---

Edit: I added a spacer character which seems to have fixed the issue, take a look now and tell me what you think.

**Comment by DylanPiercey** [comment]
I think I'm finally satisfied with this PR. However it also requires updated docs.

I was hoping ideally to also use gifs at this point. Would you mind reviewing this again and making gifs for the examples? I can then add to the PR.

File: bin/jsize [hunk 4]
```diff
@@ -7,61 +7,123 @@ const pkg = require('../package.json')
...
+const spinner = require('ora')()
 const program = require('commander')
+const logUpdate = require('log-update')
 const prettyBytes = require('pretty-bytes')
-const spinner = require('ora')('Starting...').start()
+const stringWidth = require('string-width')
+const createTable = require('text-table')
 const jsize = require('../index')
+const originalLog = console.log
+const spacer = chalk.hidden('.') // Works better with table alignment than empty space.
```
Comment (antonmedv) [inline bin/jsize]: Actually looks like this in my terminal: 
image

Comment (antonmedv) [reply bin/jsize]: Can copy-paste will be difficult

File: bin/jsize [hunk 5]
```diff
@@ -7,61 +7,123 @@ const pkg = require('../package.json')
...
-  spinner.succeed(padEnd(spinner.text, padding, ' ') + info)
+
+  // Pad table on top and bottom.
+  table.unshift(emptyRow)
+  table.push(emptyRow)
+
+  // Enable logs for long enough to print to the screen.
+  const curLog = console.log
+  console.log = originalLog
+  logUpdate(createTable(table, { stringLength: stringWidth }))
```
Comment (antonmedv) [inline bin/jsize]:
```js
logUpdate(createTable(table, { stringLength: stringWidth }).split('\n').map(s => '  ' + s).join('\n'))
```

**Comment by antonmedv** [comment]
Also what about separating to a few files in `src` dir?

**Comment by DylanPiercey** [comment]
I'll take another look into the '..' thing, what terminal are you using?

Also re separating it out I'm not sure it makes sense since this stuff is only used in the CLI

**Comment by antonmedv** [comment]
I'm using iTerm2.9.

You can pad without '..' and hidden stuff. Take a look:
```
logUpdate(createTable(table, { stringLength: stringWidth }).split('\n').map(s => '  ' + s).join('\n'))
```

**Comment by DylanPiercey** [comment]
@antonmedv I just updated the PR, it still works for me if I use 'hidden' spaces. If I don't make them hidden the alignment goes all weird in the OS X terminal. This should resolve the issues you had though.

**Comment by antonmedv** [comment]
Still don't know why you don't take my solution? 😀 It's less LOC and simple.

**Comment by DylanPiercey** [comment]
Used your solution and simplified things a bit.

**Comment by antonmedv** [comment]
Now no padding at all:
image

**Comment by DylanPiercey** [comment]
Seems to be working in every term I have?

**Review by antonmedv — APPROVED** [review]

**Pull request merged**